src/YDialogSpy.cc

```diff
@@ -514,7 +514,12 @@ void YDialogSpy::exec()
 
             continue;
         }
-        else if ( event->widget() == priv->upButton ) priv->moveSelectedUp();
+
+        // check for NULL which is generated for timeout events which can be
+        // triggered by the original dialog
+        if (!event->widget()) continue;
```

Comment:
All the following conditions compare the event widget (`event->widget()`) with created widgets (pointers) in the spy dialog. Any other dialog cannot contain these widgets, if the event is from another dialog the widgets must be different.

But it's a good idea to have this check anyway, if we add another condition later we might get into the same problem again...

Thanks Stefan!